lib/src/main/java/com/knuddels/jtokkit/Cl100kParser.java

```diff
@@ -0,0 +1,247 @@
+package com.knuddels.jtokkit;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+import static java.lang.Character.*;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.binarySearch;
+
+public class Cl100kParser {
+    private static final String SDTM = "sdtmSDTMſ";
```

Comment:
as mentioned in the first PR, funny enough `'ſ` is a valid contraction that's detected as a separate token - most likely unintended, but tiktoken seems to behave the same way:
<img width="596" alt="image" src="https://github.com/knuddelsgmbh/jtokkit/assets/1841944/84dd3138-8a23-48c8-ab8c-3d5bd9876fe2">